sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,196 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+
+#include "sw/device/lib/base/mmio.h"
+
+// TODO - Should be autogenerated.
+// - Should padctrl/pinmux share some defines/constants (such as MIO_NUM) or do
+// we just autogen the same number twice?
+#define DIF_PADCTRL_DIO_NUM 32
+#define DIF_PADCTRL_MIO_NUM 32
+#define DIF_PADCTRL_ATTR_WIDTH 8
+#define DIF_PADCTRL_ADDITIONAL_ATTR_WIDTH 2
```

Comment:
As there are 6 standard attributes DIF_PADCTRL_ADDITIONAL_ATTR_WIDTH = DIF_PADCTRL_ATTR_WIDTH - 6 (or 0 if attribute width is less than 6). I've just defined things seperately for now without this relationship as some of this will need to get autogenerated anyway.